torch/fft/__init__.py

```diff
@@ -549,3 +555,177 @@
+
+Returns Discrete Fourier Transform sample frequencies for signal size :attr:`n`.
+
+Note:
+
+    By convention, the FFT returns positive frequency terms first, followed by
+    the negative frequencies in reverse order, so that ``f[-i]`` in python
+    gives the negative frequency terms. For an FFT of length :attr:`n` and
+    with inputs spaced in length unit :attr:`d`, the frequencies are::
+
+        f = [0, 1, ..., (n - 1) // 2, -(n // 2), ..., -1] / (d * n)
+
+Note:
+    For even lengths, the Nyquist frequency at ``f[n/2]`` can be thought of as
+    either negative or positive. :func:`~torch.fft.fftfreq` follows NumPy's
+    convention of taking it to be negative.
+
+
+Args:
+    n (int): the FFT length
+    d (float, optional): The sampling length scale.
+        The unit length (or time) between individual samples in the FFT input.
+        If given, the frequencies are given in physical units of 1/unit.
+    {dtype}
+    {layout}
+    {device}
+    {requires_grad}
+
+Example:
+
+    >>> import torch.fft
+    >>> torch.fft.fftfreq(5)
+    tensor([ 0.0000,  0.2000,  0.4000, -0.4000, -0.2000])
+
+    In this example, the Nyquist frequency at ``f[2]`` is given as negative:
+    >>> torch.fft.fftfreq(4)
+    tensor([ 0.0000,  0.2500, -0.5000, -0.2500])
+
+""".format(**factory_common_args))
+
+rfftfreq = _add_docstr(_fft.fft_rfftfreq, r"""
+rfftfreq(n, d=1.0, *, dtype=None, layout=torch.strided, device=None, requires_grad=False) -> Tensor
+
+Returns sample frequencies for :func:`~torch.fft.rfft` with signal size :attr:`n`.
+
+Note:
+
+    :func:`~torch.fft.rfft` returns Hermitian one-sided output. So, only the
+    positive frequency terms are included. For a real FFT of length :attr:`n`
+    and with inputs spaced in length unit :attr:`d`, the frequencies are::
+
+        f = torch.arange((n + 1) // 2) / (d * n)
+
+Note:
+    For even lengths, the Nyquist frequency at ``f[n/2]`` can be thought of as
+    either negative or positive. Unlike :func:`~torch.fft.fftfreq`,
+    :func:`~torch.fft.rfftfreq` always considers it to be positive.
+
+
+Args:
+    n (int): the real FFT length
+    d (float, optional): The sampling length scale.
+        The unit length (or time) between individual samples in the FFT input.
+        If given, the frequencies are given in physical units of 1/unit.
+    {dtype}
+    {layout}
+    {device}
+    {requires_grad}
+
+Example:
+
+    >>> import torch.fft
+    >>> torch.fft.rfftfreq(5)
+    tensor([ 0.0000,  0.2000,  0.4000])
+
+    >>> torch.fft.rfftfreq(4)
+    tensor([ 0.0000,  0.2500, 0.5000])
+
+    Compared to the output from :func:`~torch.fft.fftfreq`, we see that the
+    Nyquist frequency at ``f[2]`` has changed sign:
+    >>> torch.fft.fftfreq(4)
+    tensor([ 0.0000,  0.2500, -0.5000, -0.2500])
+
+""".format(**factory_common_args))
+
+fftshift = _add_docstr(_fft.fft_fftshift, r"""
+fftshift(x, dim=None) -> Tensor
+
+Shift zero-frequency FFT terms to the center of the tensor.
```

Comment:
"Shifts the zero-frequency FFT terms to the center of :attr:\`input\`."

However, I don't know if this is a great one-sentence description of this function. It doesn't mention `dim`, and it uses the terms "shifts" and "center", which have non-obvious meanings. 

Note that "shifts" is preferable to "shift" because the function shifts things.

Reply:
Changed to "Reorder n-dimensional FFT output to have negative frequency terms first." 

Still not 100% happy but not sure how to improve it.